Sources/Workspace/Workspace+Manifests.swift

```diff
@@ -384,33 +384,23 @@ extension Workspace {
     }
 
     /// Returns manifest interpreter flags for a package.
-    // TODO: should this be throwing instead?
-    public func interpreterFlags(for packagePath: AbsolutePath) -> [String] {
-        do {
-            guard let manifestLoader = self.manifestLoader as? ManifestLoader else {
-                throw StringError("unexpected manifest loader kind")
-            }
+    public func interpreterFlags(for manifestPath: AbsolutePath) throws -> [String] {
+        guard let manifestLoader = self.manifestLoader as? ManifestLoader else {
+            throw StringError("unexpected manifest loader kind")
+        }
 
-            let manifestPath = try ManifestLoader.findManifest(
-                packagePath: packagePath,
-                fileSystem: self.fileSystem,
-                currentToolsVersion: self.currentToolsVersion
-            )
-            let manifestToolsVersion = try ToolsVersionParser.parse(
-                manifestPath: manifestPath,
-                fileSystem: self.fileSystem
-            )
+        let manifestToolsVersion = (try? ToolsVersionParser.parse(
+            manifestPath: manifestPath,
+            fileSystem: self.fileSystem
+        )) ?? self.currentToolsVersion
 
-            guard self.currentToolsVersion >= manifestToolsVersion || SwiftVersion.current.isDevelopment,
-                  manifestToolsVersion >= ToolsVersion.minimumRequired
-            else {
-                throw StringError("invalid tools version")
-            }
-            return manifestLoader.interpreterFlags(for: manifestToolsVersion)
-        } catch {
-            // We ignore all failures here and return empty array.
-            return []
+
+        guard self.currentToolsVersion >= manifestToolsVersion || SwiftVersion.current.isDevelopment,
+                manifestToolsVersion >= ToolsVersion.minimumRequired
+        else {
+            throw StringError("invalid tools version")
```

Comment:
nit: can you include the invalid tools version in the string to make the issue easier to debug?

Reply:
Does this error path also have corresponding test coverage here?

Reply:
I just shuffled code around and I don’t know how I would trigger this error scenario.

Reply:
Addressed in https://github.com/swiftlang/swift-package-manager/pull/8172